Dockerfile

```diff
@@ -1,7 +1,40 @@
-FROM ubuntu:latest
-WORKDIR /app
-COPY ./target/release/prover-server .
-COPY ./libzktrie.so .
-ENV LD_LIBRARY_PATH=/app/libzktrie.so:
+FROM zktachyon/halo2:jammy-latest AS builder
+LABEL maintainer="Kroma <kroma@kroma.network>"
+
+ARG RUST_TOOLCHAIN_VERSION=nightly-2022-12-10
+
+RUN apt update && \
+    apt install -y --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    curl \
+    libclang-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain ${RUST_TOOLCHAIN_VERSION} -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+RUN curl -LO "https://golang.org/dl/go1.20.3.linux-amd64.tar.gz" && \
+    rm -rf /usr/local/go && tar -C /usr/local -xzf go1.20.3.linux-amd64.tar.gz
+ENV PATH="${PATH}:/usr/local/go/bin"
+
+COPY . /usr/src/kroma-prover
+WORKDIR /usr/src/kroma-prover
+
+RUN cargo build --release --bin prover-server --features=tachyon
+RUN mkdir -p /temp-lib && \
+    cp $(find /usr/lib/ -name libtachyon.so.0) /temp-lib/ && \
+    cp $(find ./target/release/ -name libzktrie.so) /temp-lib/ && \
+    cp $(find /usr/lib/ -name libgomp.so.1) /temp-lib/ && \
+    cp $(find /usr/lib/ -name libgmpxx.so.4) /temp-lib/
```

Comment:
```suggestion
    cp $(find ./target/release/ -name libzktrie.so) /temp-lib/ 
```

Reply:
By the way, why do we need `temp-lib`?

Reply:
@chokobole libzktrie.so was under ./target/release/xxx/libzktrie.so.

I couldn't find a way to copy from builder image since xxx part is dynamic. So I used a tactic to copy to temp-lib then copy since then it's static!